File: selectionSort.py
def print_array(arr):
    for num in arr:
        print(num, end=" ")
    print()

def selection_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        index_of_min = i
        for j in range(i + 1, n):
            if arr[j] < arr[index_of_min]:
                index_of_min = j

        arr[i], arr[index_of_min] = arr[index_of_min], arr[i]

if __name__ == "__main__":
    n = int(input("Enter the number of elements: "))
    arr = []

    print("Enter the elements of the array:")
    for _ in range(n):
        element = int(input())
        arr.append(element)

    print("Array before sorting:")
    print_array(arr)

    selection_sort(arr)

    print("Array after sorting:")
    print_array(arr)

File: selectionSort.java
import java.util.*;

public class selectionSort {
    public static void printArray(int[] arr) {
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    public static void selectionSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            int indexOfMin = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[indexOfMin]) {
                    indexOfMin = j;
                }
            }

            int temp = arr[i];
            arr[i] = arr[indexOfMin];
            arr[indexOfMin] = temp;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();
        int[] arr = new int[n];

        System.out.println("Enter the elements of the array:");
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        System.out.println("Array before sorting:");
        printArray(arr);

        selectionSort(arr);

        System.out.println("Array after sorting:");
        printArray(arr);

        scanner.close();
    }
}

